spring-boot-project/spring-boot/src/test/java/org/springframework/boot/diagnostics/analyzer/FlywayMigrationScriptMissingFailureAnalyzerTests.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2012-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.diagnostics.analyzer;
+
+import org.junit.Test;
+
+import org.springframework.boot.diagnostics.FailureAnalysis;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests for {@link FlywayMigrationScriptMissingFailureAnalyzer}
+ *
+ * @author Anand Shastri
+ */
+public class FlywayMigrationScriptMissingFailureAnalyzerTests {
+
+	private final FlywayMigrationScriptMissingFailureAnalyzer analyzer = new FlywayMigrationScriptMissingFailureAnalyzer();
+
+	@Test
+	public void analysisForFlywayScriptMissingFailureWhenIllegalStateExceptionIsForFlyway() {
```

Comment:
This should change with the case where none of the locations exist and one case where no location is provided (see `FlywayAutoConfiguration`).